source/opt/folding_rules.cpp

```diff
@@ -598,7 +575,7 @@ FoldingRule MergeMulMulArithmetic() {
 }
 
 // Merges divides into subsequent multiplies if each instruction contains one
-// constant operand.
+// constant operand. Does not support integer operations.
 // Cases:
 // 2 * (x / 2) = 4 / x
```

Comment:
I know this is already in, but this should be `2 * (x / 2) = x`.  The comment below does not look right either.